darts/timeseries.py

```diff
@@ -866,7 +866,40 @@ def from_group_dataframe(
         df = df[static_cov_cols + extract_value_cols + extract_time_col]
 
         if time_col:
-            df = df.set_index(df[time_col])
+            if np.issubdtype(df[time_col].dtype, object) or np.issubdtype(
+                df[time_col].dtype, np.integer
+            ):
+                # Try to convert to integers if needed and set index
+                try:
+                    df[time_col] = df[time_col].astype(int)
+                    df = df.set_index(df[time_col])
+                except ValueError:
+                    pass
+
+            if np.issubdtype(df[time_col].dtype, np.integer):
+                # The integer conversion worked or it was already an integer
+                pass
+
+            elif np.issubdtype(df[time_col].dtype, object):
+                # The integer conversion failed; try datetimes
+                try:
+                    df.index = pd.DatetimeIndex(df[time_col])
+                    df.drop(columns=time_col)
+                except ValueError:
+                    raise_log(
+                        AttributeError(
+                            "'time_col' is of 'object' dtype but doesn't contain valid timestamps"
+                        )
+                    )
+            elif np.issubdtype(df[time_col].dtype, np.datetime64):
+                df.index = pd.DatetimeIndex(df[time_col])
+                df.drop(columns=time_col)
+            else:
+                raise_log(
+                    AttributeError(
+                        "Invalid type of `time_col`: it needs to be of either 'str', 'datetime' or 'int' dtype."
+                    )
+                )
```

Comment:
That's a nice start to handle the different types. We can actually simplify this here a bit. What we can/should assume is that if the column was a str/object or np.datetime64 then we should create a DatetimeIndex. Otherwise, it must be an integer index.

So, we can let pandas handle the exceptions in case it does not work. Then the code becomes something as below:

```suggestion
            if np.issubdtype(df[time_col].dtype, object) or np.issubdtype(
                df[time_col].dtype, np.datetime64
            ):
                df.index = pd.DatetimeIndex(df[time_col])
                df = df.drop(columns=time_col)
            else:
                df = df.set_index(time_col)
```

Also as a general tip: **try** to avoid try/except as much as possible. Especially when you don't raise the error in the `except:`. This can very quickly lead to unintended behavior.

Reply:
Completely agree, thanks 👍